tests/Doctrine/Tests/Common/Collections/ClosureExpressionVisitorTest.php

```diff
@@ -143,6 +143,17 @@ public function testWalkInComparison() : void
         self::assertTrue($closure(new TestObject('04')));
     }
 
+    public function testWalkInComparisonObjects() : void
+    {
+        $closure = $this->visitor->walkComparison($this->builder->in('foo', [new TestObject(1), new TestObject(2)]));
+
+        self::assertTrue($closure(new TestObject(new TestObject(2))));
+        self::assertTrue($closure(new TestObject(new TestObject(1))));
+        self::assertFalse($closure(new TestObject(new TestObject(0))));
+        self::assertFalse($closure(new TestObject(4)));
+        self::assertFalse($closure(new TestObject('04')));
```

Comment:
What is the point of the 2 last assertions? I might understand asserting an object is not confused with `4`, but why the need to compare it with `'04'`?

Reply:
In all honesty, I can't completely recall. I'm fairly certain they came from the original tests that I duplicated, but I'm not sure why I didn't put them into a similar object structure are the previous assertions. I can correct that if you'd prefer.

Reply:
I would prefer if everything made perfect sense for the test at hand. Not sure using a similar object structure will help though, I mean it's only interesting if you have 4 in the list where you look with the `IN` operator, and '04' as a thing you are looking for, isn't it?

Reply:
Cool, no problem. Totally agree. I think I was wanting to add a negative assertion. I'll get it squared away today.

Reply:
@greg0ire Sorry that took so long, other things came up today. Corrected the test and made it more sensible.

Reply:
Don't worry, I wasn't in a hurry :)